src/main/java/com/squadb/workassistantapi/reservation/domain/Reservation.java

```diff
@@ -0,0 +1,142 @@
+    private Member member;
+
+    @JoinColumn(name = "book_id", nullable = false)
+    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.PERSIST)
+    private Book book;
+
+    @Enumerated(EnumType.STRING)
+    private ReservationStatus status;
+
+    @Column(nullable = false)
+    private LocalDateTime reservationDate;
+
+    //대여유보기간
+    private static final int RETENTION_PERIOD_OF_RENTAL = 3;
+
+    //1인당 최대 예약 가능 개수
+    public static final int MAX_COUNT_PER_MEMBER = 3;
+
+    //한 종의 책에 대해 최대 예약 가능 개수
+    public static final int MAX_COUNT_PER_BOOK = 5;
+
+    @Builder
+    private Reservation(Member member, Book book, ReservationStatus status, LocalDateTime reservationDate) {
+        this.member = member;
+        this.book = book;
+        this.status = status;
+        this.reservationDate = reservationDate;
+    }
+
+    public static Reservation createReservation(Member member, Book book, ReservationValidator reservationValidator) {
+        validateNotNull(member, book);
+        validateBookOutOfStock(book);
+        reservationValidator.canReserve(member, book);
+        return Reservation.builder()
+                .member(member)
+                .book(book)
+                .status(ReservationStatus.WAITING)
+                .reservationDate(LocalDateTime.now())
+                .build();
+    }
+
+    private static void validateBookOutOfStock(Book book) {
+        if (!book.isOutOfStock()) {
+            throw new ReservationException(ReservationErrorCode.NOT_RESERVABLE, "대여가능한 책은 예약할 수 없습니다.");
+        }
+    }
+
+    public void cancelBy(Member member) {
+        validateNotNull(member);
+        validateReservedBy(member);
+        validateIsStatusWaiting();
+        status = ReservationStatus.CANCELED;
+    }
+
+    private void validateReservedBy(Member member) {
+        if (!isReservedBy(member)) {
+            throw new ReservationException(ReservationErrorCode.NOT_AUTHORIZED);
+        }
+    }
+
+    public boolean isReservedBy(Member targetMember) {
+        return this.member == targetMember;
+    }
+
+    public boolean canRentable() {
+        return !book.isOutOfStock();
+    }
+
+    private void validateIsStatusWaiting() {
+        if (status != ReservationStatus.WAITING) {
+            throw new ReservationException(ReservationErrorCode.ILLEGAL_STATUS);
+        }
+    }
+
+    public boolean revokeReservationExpiringOn(LocalDateTime targetDate) {
+        validateNotNull(targetDate);
+        if (isExpiringOn(targetDate)) {
+            status = ReservationStatus.REVOKED;
+            return true;
+        }
+        return false;
+    }
+
+    private boolean isExpiringOn(LocalDateTime targetDate) {
+        LocalDateTime expiryDate = getExpiryDate();
+        return expiryDate.isBefore(targetDate);
+    }
+
+    /**
+     * 대여가능날짜를 포함해 대여유보기간이 지난다면 예약이 만료된다.
+     * ex) 대어유보기간이 3일이고 대출가능 시간이 1일 15시라면 만료일은 4일 00시 이다.
+     */
+    private LocalDateTime getExpiryDate() {
```

Comment:
```suggestion
    private LocalDateTime findExpiryDate() {
```
getXXX 으로 네이밍을 짓는 건 관례적으로 getter의 의미가 강하기 때문에, find와 같은 다른 네이밍으로 짓는 게 좋을 거 같아요.

Reply:
그것까진 생각을 못했네요.. 감사합니다!
(넥스트스텝을 빨리 들어야겠네요 ㅋㅋㅋ)